#!/usr/bin/env python
PACKAGE = "sl_crazyflie_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("KP_thrust",    double_t,    0, "An Integer parameter", 0.6,  0, 2)
gen.add("KI_thrust",    double_t,    0, "An Integer parameter", 1.2,  0, 2)
gen.add("KD_thrust",    double_t,    0, "An Integer parameter", 0.,  0, 2)

gen.add("KP_climb_rate",   double_t,    0, "An Integer parameter", 1,  0, 4)
gen.add("KI_climb_rate",    double_t,    0, "An Integer parameter", 0,  0, 20)
gen.add("KD_climb_rate",    double_t,    0, "An Integer parameter", 0,  0, 20)

gen.add("KP_xy",   double_t,    0, "An Integer parameter", 22.5,  0, 30)
gen.add("KI_xy",    double_t,    0, "An Integer parameter", 20,  0, 60)
gen.add("KD_xy",    double_t,    0, "An Integer parameter", 0,  0, 20)

gen.add("KP_xy_speed",   double_t,    0, "An Integer parameter", 1.2,  0, 4)
gen.add("KI_xy_speed",    double_t,    0, "An Integer parameter", 0,  0, 2)
gen.add("KD_xy_speed",    double_t,    0, "An Integer parameter", 0,  0, 2)

gen.add("pitch_roll_cap",    double_t,    0, "An Integer parameter", 45,  0, 60)
gen.add("max_xy_error",    double_t,    0, "An Integer parameter", 1.5,  0, 4)

gen.add("max_altitude_error",    double_t,    0, "An Integer parameter", 2.,  0, 4)
gen.add("Nom_thrust",    double_t,    0, "An Integer parameter", 0.7,  0, 1.0)

#gen.add("KP_thrust",    double_t,    0, "An Integer parameter", 0.6,  0, 2)
#gen.add("KI_thrust",    double_t,    0, "An Integer parameter", 1.2,  0, 2)
#gen.add("KD_thrust",    double_t,    0, "An Integer parameter", 0.,  0, 2)

#gen.add("KP_climb_rate",   double_t,    0, "An Integer parameter", 1.4,  0, 4)
#gen.add("KI_climb_rate",    double_t,    0, "An Integer parameter", 1.6,  0, 20)
#gen.add("KD_climb_rate",    double_t,    0, "An Integer parameter", 0,  0, 20)

#gen.add("KP_xy",   double_t,    0, "An Integer parameter", 20.4,  0, 30)
#gen.add("KI_xy",    double_t,    0, "An Integer parameter", 24.6,  0, 60)
#gen.add("KD_xy",    double_t,    0, "An Integer parameter", 0,  0, 20)

#gen.add("KP_xy_speed",   double_t,    0, "An Integer parameter", 0.68,  0, 4)
# gen.add("KI_xy_speed",    double_t,    0, "An Integer parameter", 0.2,  0, 2)
# gen.add("KD_xy_speed",    double_t,    0, "An Integer parameter", 0.08,  0, 2)
#
# gen.add("pitch_roll_cap",    double_t,    0, "An Integer parameter", 20,  0, 60)
# gen.add("max_xy_error",    double_t,    0, "An Integer parameter", 1.5,  0, 4)
#
# gen.add("max_altitude_error",    double_t,    0, "An Integer parameter", 2.,  0, 4)
# gen.add("Nom_thrust",    double_t,    0, "An Integer parameter", 0.7,  0, 1.0)

exit(gen.generate(PACKAGE, "sl_crazyflie_controller", "pid_cfg"))
